app/scripts/models/import.js

```diff
@@ -1,178 +1,219 @@
-              currentUser.id = evaluation.creator;
-            }
-            evaluation.creator = currentUser;
-            var foundUser = false;
-            // Find the first Person that matches the ID of the current user
-            results.forEach(function (data) {
-              if (!foundUser && data.type === 'Person' &&
-                    data.id === currentUser.id) {
-                // overwrite the current user with the new data
-                angular.extend(currentUser, data);
-                foundUser = true;
+          importModel.fromExpanded(expanded);
+        });
+      },
+
+      fromExpanded: function (evalData) {
+        evalData
+          .forEach(
+            compactEach(function (results) {
+              var evaluation = results
+                .reduce(function (result, data) {
+                  if (data.type === 'Evaluation') {
+                    if (typeof result !== 'undefined') {
+                      throw new Error('Only one evaluation object allowed in JSON data');
+                    }
+
+                    return data;
+                  }
+
+                  return result;
+                }, undefined);
+
+              if (!evaluation) {
+                throw new Error('No evaluation found in data');
               }
-            });
 
-            // Take all data that isn't the evaluation or the current user
-            evaluation.otherData = results.reduce(function (otherData, data) {
-              if (data !== evaluation && data.id !== currentUser.id) {
-                otherData.push(data);
+              // If the creator has an id, give that id to the current user
+              if (typeof evaluation.creator === 'string' &&
+                evaluation.creator.indexOf('_:') === 0) {
+                currentUser.id = evaluation.creator;
               }
-              return otherData;
-            }, [currentUser]);
-
-            if (evaluation.lang) {
-              // This is a workaround for what seems to be a bug in the
-              // JSON-LD lib. It outputs ['e', 'n'] instead of 'en', so we
-              // join to fix this.
-              if (angular.isArray(evaluation.lang)) {
-                evaluation.lang = evaluation.lang.join('');
+
+              evaluation.creator = currentUser;
+
+              var foundUser = false;
+
+              // Find the first Person that matches the ID of the current user
+              results
+                .forEach(function (data) {
+                  if (
+                    !foundUser &&
+                    data.type === 'Person' &&
+                    data.id === currentUser.id
+                  ) {
+                  // overwrite the current user with the new data
+                    angular.extend(currentUser, data);
+                    foundUser = true;
+                  }
+                });
+
+              // Take all data that isn't the evaluation or the current user
+              evaluation.otherData = results
```

Comment:
this can be a `filter`

Reply:
Ah I see now,

even better, this and some other finding methods can be replaced by the json-ld API.

I have been struggling a lot the past few weeks to get a good grip on json-ld and I think a lot of the data importing and exporting can be refined with the use of json-ld framing for example.

I have not looked into how this affects performance etc, but I like to add more json-ld API usage since what we are importing and exporting is in this data format.